lib/features/modeling/behavior/ResizeBehavior.js

```diff
@@ -160,10 +160,25 @@ export function getParticipantResizeConstraints(laneShape, resizeDirection, bala
 
     var otherTrbl = asTRBL(other);
 
-    if (/n/.test(resizeDirection)) {
-      if (isHorizontalLane && otherTrbl.top < (laneTrbl.top - 10)) {
+    // lane flags
+    if (isHorizontalLane) {
+      if (otherTrbl.top < (laneTrbl.top - 10)) {
+        isFirst = false;
+      }
+      if (otherTrbl.bottom > (laneTrbl.bottom + 10)) {
+        isLast = false;
+      }
+    }
+    else if (!isHorizontalLane) {
```

Comment:
I believe this is duplicated, right? If one branch checks for `isHorizontalLane` then the other one naturally checks for the opposite (`!isHorizontalLane`)?

```suggestion
    else {
```

Reply:
Yes, good catch. I'll update it.